src/components/CopyLink.js

```diff
@@ -3,43 +3,68 @@ import { Button } from "reactstrap";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faLink } from "@fortawesome/free-solid-svg-icons";
 import * as qs from 'qs';
+import PopupDialog from "./PopupDialog";
 
 const UNCLICKED_TEXT = " Copy link to data";
 const CLICKED_TEXT = " Copied link!";
 
+// uses Clipboard API to write text to clipboard
 export const writeToClipboard = (text) => {
   navigator.clipboard.writeText(text);
 };
 
-
 // For testing purposes
 let copyCallback = writeToClipboard;
+
+// sets value of copyCallback
 export const setCopyCallback = (callback) => (copyCallback = callback);
+
 export function CopyLink(props) {
   // Button to copy a link with viewTarget to the data selected
-
   const [text, setText] = useState(UNCLICKED_TEXT);
+  const [dialogLink, setDialogLink] = useState(undefined);
 
   const handleCopyLink = () => {
+    // open popup
+    setOpen(!open);
```

Comment:
Isn't this going to make the dialog always open, even if `navigator.clipboard` works fine?